src/hotspot/cpu/aarch64/sharedRuntime_aarch64.cpp

```diff
@@ -1171,7 +1171,7 @@ static void verify_oop_args(MacroAssembler* masm,
         VMReg r = regs[i].first();
         assert(r->is_valid(), "bad oop arg");
         if (r->is_stack()) {
-          __ ldr(temp_reg, Address(sp, r->reg2stack() * VMRegImpl::stack_slot_size));
+          __ ldr(temp_reg, Address(sp, (uint64_t)r->reg2stack() * VMRegImpl::stack_slot_size));
```

Comment:
Why is the cast being added here?  Is it because the multiply can overflow an int, and this is really fix for a bug that is distinct from the cleanup in this PR?  There are a couple more like this later in this file.

Reply:
This was here to match one of the Address constructors that I had.  With the template, it's no longer needed.

Reply:
I think I didn't revert this.  Retesting.